src/codegen/mod.rs

```diff
@@ -387,7 +532,13 @@ fn gen_exec(
 
     let mut input: Vec<_> = name_to_exprs
         .iter()
-        .map(|(_, pl_expr)| gen_expr(pl_expr, &mut HashMap::new(), &mut HashMap::new(), comp_unit))
+        .map(|(_, pl_expr)| {
+            // TODO is this right?
+            match gen_expr(pl_expr, &mut HashMap::new(), &mut HashMap::new(), comp_unit, label.clone()) {
+                CheckedExpr::Expr(expr) => expr,
+                CheckedExpr::ExprIdxCheck(_, expr) => expr,
```

Comment:
I think checks have to be generated for input expressions as well.

E.g.,

```rust
for block_group
with inputs[i]
do ...
```

Reply:
- [ ] gen checks here

Reply:
not very difficult, changes are pushed